### PR TITLE
Workflow update

### PR DESCRIPTION
**Pull request by v-fearam**
Adding gitignore
Moving to net8
Updating libraries
Changing ServiceBus library

### CODE REVIEW

File: src/shipping/workflow/Fabrikam.Workflow.Service.Tests/Fabrikam.Workflow.Service.Tests.csproj [hunk 1]
```diff
@@ -1,21 +1,21 @@
...
-    <PackageReference Include="Microsoft.Extensions.Logging.Debug" Version="5.0.0" />
-    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="16.10.0" />
-    <PackageReference Include="Moq" Version="4.16.1" />
-    <PackageReference Include="xunit" Version="2.4.1" />
-    <PackageReference Include="xunit.runner.visualstudio" Version="2.4.3">
+    <PackageReference Include="Microsoft.AspNetCore.TestHost" Version="8.0.0" />
+    <PackageReference Include="Microsoft.Extensions.Logging.Debug" Version="8.0.0" />
+    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="17.8.0" />
+    <PackageReference Include="Moq" Version="4.20.70" />
+    <PackageReference Include="xunit" Version="2.6.2" />
```
Comment (ferantivero) [inline src/shipping/workflow/Fabrikam.Workflow.Service.Tests/Fabrikam.Workflow.Service.Tests.csproj:13]: there are newer version for xunit. You can consider not updating them as they are minor.

File: src/shipping/workflow/Fabrikam.Workflow.Service/WorkflowService.cs [hunk 2]
```diff
@@ -4,18 +4,19 @@
...
 using System.Threading;
 using System.Threading.Tasks;
-using Microsoft.Azure.ServiceBus;
 using Microsoft.Extensions.Hosting;
 using Microsoft.Extensions.Logging;
 using Microsoft.Extensions.Options;
 using Fabrikam.Workflow.Service.Models;
 using Fabrikam.Workflow.Service.RequestProcessing;
+using Azure.Messaging.ServiceBus;
+using System.Text.Json;
```
Comment (ferantivero) [inline src/shipping/workflow/Fabrikam.Workflow.Service/WorkflowService.cs:15]: consider sort & order the usings

File: src/shipping/workflow/Fabrikam.Workflow.Service/WorkflowService.cs [hunk 3]
```diff
@@ -25,114 +26,103 @@ internal class WorkflowService : IHostedService
 
         private readonly ILogger<WorkflowService> _logger;
         private readonly IRequestProcessor _requestProcessor;
-        private readonly Func<IOptions<WorkflowServiceOptions>, IQueueClient> _createQueueClient;
+        private readonly Func<IOptions<WorkflowServiceOptions>, ServiceBusClient> _createServiceBusClient;
         private readonly IOptions<WorkflowServiceOptions> _options;
-        private IQueueClient _receiveClient;
```
Comment (ferantivero) [inline src/shipping/workflow/Fabrikam.Workflow.Service/WorkflowService.cs:30]: could you please detail a bit why we are changing this implementation?

**Review by ferantivero — CHANGES_REQUESTED** [review]

File: src/shipping/workflow/Fabrikam.Workflow.Service/WorkflowService.cs (hunk 3 above)
Comment (v-fearam) [reply src/shipping/workflow/Fabrikam.Workflow.Service/WorkflowService.cs:30]: [Microsoft.Azure.ServiceBus](https://www.nuget.org/packages/Microsoft.Azure.ServiceBus) is deprecated and it is suggested change to [Azure.Messaging.ServiceBus](https://www.nuget.org/packages/Azure.Messaging.ServiceBus/).
We implementing following the new library

File: src/shipping/workflow/Fabrikam.Workflow.Service/WorkflowService.cs (hunk 2 above)
Comment (v-fearam) [reply src/shipping/workflow/Fabrikam.Workflow.Service/WorkflowService.cs]: done

File: src/shipping/workflow/Fabrikam.Workflow.Service.Tests/Fabrikam.Workflow.Service.Tests.csproj (hunk 1 above)
Comment (v-fearam) [reply src/shipping/workflow/Fabrikam.Workflow.Service.Tests/Fabrikam.Workflow.Service.Tests.csproj]: done

**Pull request merged**